core/capabilities/ccip/common/crcwconfig.go

```diff
@@ -0,0 +1,80 @@
+package common
+
+import (
+	"context"
+	"fmt"
+
+	chainsel "github.com/smartcontractkit/chain-selectors"
+
+	cciptypes "github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
+	"github.com/smartcontractkit/chainlink-common/pkg/loop"
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+// GetChainReaderParams is a struct that contains the parameters for GetChainReader.
+type GetChainReaderParams struct {
+	Lggr             logger.Logger
+	Relayer          loop.Relayer
+	ChainID          string
+	DestChainID      string
+	HomeChainID      string
+	Ofc              OffChainConfig
+	ChainSelector    cciptypes.ChainSelector
+	RelayChainFamily string
+}
+
+// GetChainWriterParams is a struct that contains the parameters for GetChainWriter.
+type GetChainWriterParams struct {
+	ChainID               string
+	Relayer               loop.Relayer
+	Transmitters          map[types.RelayID][]string
+	ExecBatchGasLimit     uint64
+	ChainFamily           string
+	OfframpProgramAddress []byte
+}
+
+// GetChainReaderWriter is an interface that defines the methods to get a ContractReader and a ContractWriter.
+type GetChainReaderWriter interface {
+	GetChainReader(ctx context.Context, params GetChainReaderParams) (types.ContractReader, error)
+	GetChainWriter(ctx context.Context, params GetChainWriterParams) (types.ContractWriter, error)
+}
+
+// CRCW is a struct that implements the GetChainReaderWriter interface for all chains.
+type CRCW struct {
+	EVMCRCW    GetChainReaderWriter
+	SolanaCRCW GetChainReaderWriter
+	Lggr       logger.Logger
+}
```

Comment:
```suggestion
type MultiChainRW struct {
	EVM    GetChainReaderWriter
	Solana GetChainReaderWriter
	Lggr     logger.Logger
}
```

Is the `Lggr` used anywhere?